framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceImplementation.java

```diff
@@ -1001,6 +1003,33 @@ protected void updateAnnotationSet(
       return;
     }
 
+    // For parameters, If rhsmATM is Nullable or MonotonicNonNull and has the
+    // UnknownInitialization annotation, and if the annotation is not
+    // UnknownInitialization(java.lang.Object.class), replace it with
+    // UnknownInitialization(java.lang.Object.class). This is because there is
+    // likely a constructor where it hasn't been initialized, and we haven't
+    // considered its effect. Otherwise, WPI might get stuck in a loop.
+    if (defLoc == TypeUseLocation.PARAMETER
```

Comment:
Why is this only an issue for parameters? I'm especially surprised that you need to do this for parameters but not for fields - the underlying issue seems to be about fields.

Reply:
It used to be for all assignments and pseudo-assignments, but to narrow down the possible loss of information, I thought it might be better to merely block the propagation of potentially incorrect annotations to other places by filtering the parameters. This approach also stopped the loop problem in my test cases, just like the previous approach. However, to ensure accuracy, I am refining the entire approach, changing its application to field assignments, and transferring it to the Nullness Checker. After my tests pass, I will commit the changes. Thank you very much for your valuable feedback.